src/agent/coverage/src/block.rs

```diff
@@ -104,6 +106,68 @@ impl CommandBlockCov {
     pub fn try_from_report(report: BlockCoverageReport) -> Result<Self> {
         Self::try_from(report)
     }
+
+    pub fn source_coverage(&self, debuginfo: &mut DebugInfo) -> Result<SourceCoverage> {
```

Comment:
Given a debug info handler, we can now convert in-memory block coverage to JSON-serializable source coverage.